src/test/scala/com/crealytics/spark/excel/IntegrationSuite.scala

```diff
@@ -83,4 +84,40 @@ class IntegrationSuite extends FunSuite with PropertyChecks with DataFrameSuiteB
       assertDataFrameEquals(expected, result)
     }
   }
+
+  test("handles null values correctly") {
+    forAll(rowsGen, MinSuccessful(20)) { rows =>
+      val expected = spark.createDataset(rows).toDF
+
+      // We need two dataframes, one with null values, one with empty strings, this is because we want ExcelFileSaver to
+      // write an empty string, if there's a null in that column. expectedWithEmptyStr is what the dataframe should look
+      // like when the Excel spreadsheet is saved
+      val expectedWithNull = expected.withColumn("aString", lit(null: String))
+      // Generate the same DataFrame but with empty strings
+      val expectedWithEmptyStr = expected.withColumn("aString", lit("": String))
+      // Set the schema so that aString is nullable
+      expectedWithEmptyStr.schema.fields.update(6, new StructField("aString", DataTypes.StringType, true))
+
+      val tempFile: File = File.createTempFile("spark_excel_null_test_", ".xlsx")
+      val fileName = tempFile.getAbsolutePath
+
+      expectedWithNull.write
```

Comment:
When we convert a excel file into a dataframe, if the first row contains null in the last column, it is excluding the entire column. Is there any way to fix this? Thanks.

Reply:
Can you create a new issue for this?